social/app/models/node.py

```diff
@@ -34,27 +35,62 @@ class Node(models.Model):
     def __str__(self):
         return '%s (%s; %s)' % (self.name, self.host, self.service_url)
 
-    def _get_author(self, uuid):
-        url = urlparse.urljoin(self.service_url, 'author/' + str(uuid))
-        return requests.get(url, auth=(self.username, self.password))
+    def _get_author(self, author_id):
+        url = urlparse.urljoin(self.service_url, "author/" + str(author_id))
+        return requests.get(url, auth=self.auth())
+
+    def auth(self):
+        return self.username, self.password
+
+    def _get_post(self, post_id):
+        url = urlparse.urljoin(self.service_url, "posts/" + str(post_id))
+        return requests.get(url, auth=self.auth())
+
+    def get_author(self, author_id):
+        return self._get_author(author_id).json()
+
+    def get_post(self, post_id):
+        return self._get_post(post_id).json()
+
+    def get_post_comments(self, post_uuid):
+        """
+        Returns a list of dicts that represents all of the Comments fetched for a particular remote Post,
+        traversing pagination if required.
+        """
+        base_url = urlparse.urljoin(self.service_url, "posts/%s/comments" % str(post_uuid))
+        json = requests.get(base_url, auth=self.auth()).json()
+
+        all_comments = json["comments"]
+
+        while "next" in json:
+            # Depending on how the other server interpreted the spec, a lack of a next page is rendered as either
+            # the next field not existing, or the next field being set to an empty value, so we gotta check for both
+            next_url = json["next"]
+            if not next_url:
```

Comment:
I've done a stronger check of the `next` value in my code [here](https://github.com/TeamAADGT/CMPUT404-project-socialdistribution/pull/115/files#diff-06671048831a033f791f24f59173b8b7R108) that also does a URL validation.